### ISSUE TITLE
Missing cost estimate for `azurerm_postgresql_flexible_server`

### ISSUE THREAD

**Issue by MHarutunian** [infracost/infracost#1884]
Hi,
the breakdown for `azurerm_postgresql_flexible_server` seems to have broken, it currently reports `$0.00` for all tiers (and possibly all SKU names in general). I tested various infracost versions (all `linux-amd64`), and it was still working in `v0.9.24`, but not in any of the `v0.10.x` versions I tested (including v0.10.0, v0.10.4, v0.10.7 and v0.10.8 which was just released).

Output in v0.9.24 (Expected):
```
 Name                                             Monthly Qty  Unit            Monthly Cost 
                                                                                            
 azurerm_postgresql_flexible_server.infracost_test                                              
 ├─ Compute (B_Standard_B1ms)                             730  hours                 $14.53 
 ├─ Storage                                                32  GB                     $4.38 
 └─ Additional backup storage                   Monthly cost depends on usage: $0.00 per GB 
                                                                                            
 OVERALL TOTAL                                                                       $18.91 
```

Output in v0.10.x (Actual):
```
 Name                                             Monthly Qty  Unit            Monthly Cost 
                                                                                            
 azurerm_postgresql_flexible_server.infracost_test                                              
 ├─ Compute (B_Standard_B2ms)                             730  hours                  $0.00 
 ├─ Storage                                                32  GB                     $0.00 
 └─ Additional backup storage                   Monthly cost depends on usage: $0.00 per GB 
                                                                                            
 OVERALL TOTAL                                                                        $0.00 
```

Minimal working example:
```
provider "azurerm" {
  features {}
}

resource "azurerm_postgresql_flexible_server" "infracost_test" {
  name                   = "infracost-test"
  resource_group_name    = "anything"
  location               = "East US"
  version                = "13"
  sku_name               = "B_Standard_B1ms"
  storage_mb             = 32768
  backup_retention_days  = 7
  administrator_login    = "psql-admin"
  administrator_password = "secret-value"
}
```

If there's any additional info I can provide, please let me know. Thanks!

**Comment by hugorut** [comment]
Thanks @MHarutunian, for bringing this to our attention and providing a detailed way to replicate it. We'll look at this as soon as we can.

**Comment by vdmgolub** [comment]
Thank you for the TF example! I had a quick look and it seems the issue is with the `location` attribute being specified as a string instead of a reference, it is not detected. The fix would be changing this line https://github.com/infracost/infracost/blob/master/internal/providers/terraform/azure/postgresql_flexible_server.go#L20 to
```diff
-	region := d.Get("location").String()
+	region := lookupRegion(d, []string{})
```

**Comment by hugorut** [comment]
good spot @vdmgolub, this should be an easy fix, and hopefully out in the next release. Thanks, @MHarutunian 👍

**Comment by MHarutunian** [comment]
Awesome, thank you very much for the quick replies and action :slightly_smiling_face:

**Comment by alikhajeh1** [comment]
@hugorut any update on this?

**Comment by hugorut** [comment]
@alikhajeh1 we have this in our backlog, but it's not in the current sprint. So won't be done for ~2 weeks.

**Comment by hugorut** [comment]
Hey @MHarutunian, we've merged a fix for this, and it's sitting in master. The fix will go out with the next release. We'll ping you here when it's available. Thanks.

**Comment by tim775** [comment]
@MHarutunian the fix is live in v0.10.10, cheers.

**Comment by MHarutunian** [comment]
Cheers, thank you very much! Can confirm it's working now.
